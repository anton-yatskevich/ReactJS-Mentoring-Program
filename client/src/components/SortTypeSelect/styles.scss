.sort-type__button {
	margin-left: 20px;
	background: transparent;
	border: none;
	font-size: 18px;
	font-weight: 600;
	color: rgb(77, 77, 77);

	&--active {
		color: rgb(160, 0, 0);
	}
}