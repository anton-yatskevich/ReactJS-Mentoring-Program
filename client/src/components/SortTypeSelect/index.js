import React from 'react';
import PropTypes from 'prop-types';
import noop from 'lodash.noop';

import Button from '../Button';
import './styles.scss';

const SortTypeSelect = ({ sortField, setSortField }) => {
    const sortTypes = [
        { text: 'release date', value: 'release_date', id: 'unique1' },
        { text: 'rating', value: 'rating', id: 'unique2' }
    ];

    return (
        <div>
            <span>Sort by:</span>
            {
                sortTypes.map(sortType => (
                    <Button
                        key={sortType.id}
                        onClickHandler={setSortField}
                        value={sortType.value}
                        textValue={sortType.text}
                        isActive={sortType.value === sortField}
                        className="sort-type__button"
                    />
                ))
            }
        </div>
    );
};

SortTypeSelect.propTypes = {
    sortField: PropTypes.string,
    setSortField: PropTypes.func
};

SortTypeSelect.defaultProps = {
    sortField: '',
    setSortField: noop
};

export default SortTypeSelect;
