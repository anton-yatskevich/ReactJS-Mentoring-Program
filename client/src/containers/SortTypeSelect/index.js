import { connect } from 'react-redux';
import SortTypeSelect from '../../components/SortTypeSelect';
import setSortField from '../../actions/setSortField';

function mapStateToProps({ searchParams }) {
    return {
        sortField: searchParams.sortField
    };
}

export default connect(mapStateToProps, { setSortField })(SortTypeSelect);
