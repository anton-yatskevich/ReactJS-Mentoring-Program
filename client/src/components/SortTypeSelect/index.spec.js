import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import SortTypeSelect from './index';

describe('SortTypeSelect component', () => {
    it('should be rendered with default props', () => {
        const SortTypeSelectComponent = shallow(<SortTypeSelect />);
        expect(toJson(SortTypeSelectComponent)).toMatchSnapshot();
    });
});
