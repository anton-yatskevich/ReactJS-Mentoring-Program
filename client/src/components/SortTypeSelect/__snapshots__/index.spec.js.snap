// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortTypeSelect component should be rendered with default props 1`] = `
<div>
  <span>
    Sort by:
  </span>
  <Button
    className="sort-type__button"
    isActive={false}
    key="4b2cc020-5c6f-11e9-b5bd-f1f2c4194524"
    onClickHandler={[Function]}
    textValue="release date"
    value="release date"
  />
  <Button
    className="sort-type__button"
    isActive={false}
    key="4b2cc021-5c6f-11e9-b5bd-f1f2c4194524"
    onClickHandler={[Function]}
    textValue="rating"
    value="rating"
  />
</div>
`;
